### ISSUE TITLE
Docker pull: always and state: reloaded no longer ignores unnamed containers

### ISSUE THREAD

**Issue by MrMMorris** [ansible/ansible-modules-core#3219]
##### Issue Type:
- Bug Report
##### Plugin Name:

Docker
##### Ansible Version:

**Worked:**

```
ansible 1.9.4
  configured module search path = None
```

**Broken:**

```
ansible 2.0.0.1
  config file = /Users/mmorris/Work/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### Ansible Configuration:

```
[defaults]
host_key_checking=False
display_skipped_hosts=False
force_handlers = True
hostfile = inventory/ec2.py
retry_files_enabled = False

[ssh_connection]
pipelining=True
```
##### Environment:

Ubuntu 14.04 from OSX 10.10
##### Summary:

Something has changed in the docker module so that if you have `pull: always` and `state: reloaded` with a unnamed container, it will keep on creating new containers instead of realizing it's the same container, just differently named.
##### Steps To Reproduce:

``` yaml
- name: create redis container
  docker:
    image: "redis:3.0.3"
    pull: always
    state: reloaded
```
##### Expected Results:

When a container already exists and it has all the same settings except the dynamically assigned name is different, nothing should happen:

**First run:**

```
GATHERING FACTS ***************************************************************
ok: [x.x.x.x]

TASK: [docker | create redis container] *********************
changed: [x.x.x.x]
```

**Second run:**

```
GATHERING FACTS ***************************************************************
ok: [x.x.x.x]

TASK: [docker | create redis container] *********************
ok: [x.x.x.x]
```

```
$docker ps | grep redis | awk '{print $2 "," $12}'
redis:3.0.3,sad_tesla
```
##### Actual Results:

It will create a new, separate container every time:

**First run:**

```
GATHERING FACTS ***************************************************************
ok: [x.x.x.x]

TASK: [docker | create redis container] *********************
changed: [x.x.x.x]
```

**Second run:**

```
GATHERING FACTS ***************************************************************
ok: [x.x.x.x]

TASK: [docker | create redis container] *********************
changed: [x.x.x.x]
```

```
$ docker ps | grep redis | awk '{print $2 "," $12}'
redis:3.0.3,sad_tesla
redis:3.0.3,mad_kirch
```

Not sure why this change was made, but it is very much NOT what should be happening and will result in problems for what I know are many people who use the module like this.

**Comment by MrMMorris** [comment]
Tracked down the change that broke it: https://github.com/ansible/ansible-modules-core/commit/e96549c95d44120d885bcacfeacd3d6a56fce579

Specifically:

``` python
-                command_matches = (not command or running_command.endswith(command))
+                command_matches = command == details['Config']['Cmd']
```

Pinging @AndrewPashkin for possible insight

Really weird, because I am not setting any command

**Comment by AndreiPashkin** [comment]
I guess, this is because `details['Config']['Cmd']` equals an empty list and `command` equals `None`?

**Comment by MrMMorris** [comment]
This also was causing a problem I was seeing where named containers were redeployed on every run instead of only when something changed.

**Comment by AndrewPashkin** [comment]
So it happens only when the `command` is not set?

**Comment by MrMMorris** [comment]
I can confirm this also redeploys named containers every time:

``` yaml
- docker:
    image: redis:3.0.7
    name: redis-test
    state: reloaded
    pull: always
```

@AndrewPashkin do you need me to find a fix or do you know what would work?

Edit: It looks like the redeploys with named containers is not fixed by reverting the previously mentioned change... Someone broke it somewhere else I guess :cry: Will create a new issue.

**Comment by ansibot** [comment]
@olsaki, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@cove, @softzilla, @zfil, @ThomasSteinbach, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@cove, @softzilla, @zfil, @ThomasSteinbach, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by ansibot** [comment]
This issue was migrated to https://github.com/ansible/ansible/issues/29855
